### PR TITLE
Use MQTT will for availability

### PR DESCRIPTION
**Pull request by Nilaik**
* change the will's topic name from will to available
* send offline as last will
* when sending all system parameters also send online for the will topic

### CODE REVIEW

**Comment by Nilaik** [comment]
The last will was already implemented for the topic will.

I created a pull request where I changed the will topic from "will" to "available". Also instead of sending "exit" I send "offline" as payload. In order to indicate the online status the payload "online" is send in the function writeSysParamsToMQTT.

Not sure if this is an allowed change since this would be a breaking change for version 3.1.

File: src/rancilio-pid.cpp [hunk 1]
```diff
@@ -1832,7 +1832,7 @@ void setup() {
         }
 
         if (MQTT == 1) {
-            snprintf(topic_will, sizeof(topic_will), "%s%s/%s", mqtt_topic_prefix, hostname, "will");
+            snprintf(topic_will, sizeof(topic_will), "%s%s/%s", mqtt_topic_prefix, hostname, "available");
```
Comment (LoQue90) [inline src/rancilio-pid.cpp:1835]: snprintf(topic_will, sizeof(topic_will), "%s%s/%s", mqtt_topic_prefix, hostname, "status");
I would change the topic name.
Compared to other mqtt stuff that i use, following would make more sense to me:
Topic will be "status", payload will be "online" or "offline"
Logic in my mind: If i want to get the availability i check the status of the mashine online or offline.
If the status of the mashine is online it is available.

File: src/rancilio-pid.cpp [hunk 2]
```diff
@@ -2318,6 +2318,9 @@ void writeSysParamsToMQTT(void) {
         previousMillisMQTT = currentMillisMQTT;
 
         if (mqtt.connected() == 1) {
+            // available topic (will sets it to offline)
```
Comment (LoQue90) [inline src/rancilio-pid.cpp:2321]: // status topic (will sets it to offline)

File: src/rancilio-pid.cpp [hunk 3]
```diff
@@ -2318,6 +2318,9 @@ void writeSysParamsToMQTT(void) {
...
+            mqtt_publish("available", "online");
```
Comment (LoQue90) [inline src/rancilio-pid.cpp:2322]: mqtt_publish("status", "online");

**Review by LoQue90 — CHANGES_REQUESTED** [review]
Works fine for me, would like to see a merge to master with my suggestions.
Just have to see on Home Assistant side how to get a nice readout for the temperature when stauts=offline

File: src/rancilio-pid.cpp (hunk 1 above)
Comment (Nilaik) [reply src/rancilio-pid.cpp]: Sounds good to me. I changed the topic name of the will to status and tested it with my setup.

Comment (Nilaik) [reply src/rancilio-pid.cpp]: In Home Assistant I have defined my temperature sensor as following:

```
  - name: Rancilio Silvia Temperature
    unique_id: sensor.silvia_temp
    availability:
      - payload_available: online
        payload_not_available: offline
        topic: espresso/rancilio.silvia/status
    state_topic: espresso/rancilio.silvia/temperature
    state_class: measurement
    device_class: temperature
    unit_of_measurement: °C
```

After switching my silvia off home assistant receives the will message and marks the sensor as unavailable.

**Review by LoQue90 — APPROVED** [review]
I would say fine fore merge.
Thank you!

**Pull request merged**